src/main/java/org/traccar/database/DeviceManager.java

```diff
@@ -116,13 +116,20 @@ public Device getByUniqueId(String uniqueId) throws SQLException {
     }
 
     @Override
-    public String getDevicePassword(long id, String defaultPassword) {
+    public String getDevicePassword(long id, String protocol, String defaultPassword) {
 
-        String password = getById(id).getString(Command.KEY_DEVICE_PASSWORD);
+        String password = lookupAttributeString(id, Command.KEY_DEVICE_PASSWORD, null, false);
```

Comment:
This still checks server-wide settings for the password.

Reply:
Yes, this way we have in descending order of preference:
- a device-specific web configurable value
- a group-wide web configurable value
- a server-wide web configurable value
- a protocol-wide file `(traccar.xml)` configurable  value
- a protocol-wide hard-coded default

I think it is very flexible, and I thought we agreed on that from here:
https://github.com/traccar/traccar/pull/4355#discussion_r304123522

Maybe there's something I misunderstood, please, correct me if that's the case.
Thank you.

Reply:
I don't want it to be server-wide settings configurable because server confirm is public. Anyone can see it.

Reply:
Anyway, if there's no server-wide password configured, it won't be used. Am I right? Thanks.

Reply:
I wrote before seeing your comment, but, isn't that the solution?

Reply:
I don't think we should allow it because not everyone realizes that server attributes are public.

Reply:
I see your point, and totally agree. So, are you saying that we can keep every step in this lookup for the attribute, except the server-wide attribute? Thanks.

Reply:
Yes. Or just use option with config with protocol prefix.

Reply:
Would you like if we make optional the lookup for server attributes with an additional argument to the private method `lookupAttribute` of `DeviceManager` class?

Reply:
I guess it makes sense.

Reply:
Let's do it. I'll make another PR. Thanks.